lib/bank_account/account.rb

```diff
@@ -0,0 +1,41 @@
+module BankAccount
+
+		class Account
+
+		attr_accessor :id, :initial_balance, :current_balance
+		
+		def initialize(id, initial_balance)
+			@id = id
+			@initial_balance = initial_balance
+			@current_balance = initial_balance	
+
+			if @initial_balance < 0 
+				raise ArgumentError, "an error has occurred. Balance must equal more than $0."
+			end	
+		end
+
+		def withdraw(w_amount)
+
+			if @current_balance <= w_amount
+				raise ArgumentError, " I'm sorry. You do not have enough money to complete this transaction. Please try again and/or make a deposit. Thanks. \n #{@current_balance} is currently avaialble."
+			else  @current_balance -= w_amount
+				puts "Your new balance is #{@current_balance}."
+				return @current_balance
+			end
```

Comment:
This method is really tight! I love that you checked the value first and dealt with the error right away. This is an efficient method!